sdk/program/src/stake/state.rs

```diff
@@ -227,6 +227,13 @@ impl StakeStateV2 {
         }
     }
 
+    pub fn delegation_ref(&self) -> Option<&Delegation> {
```

Comment:
nit for the future: I don't think it's worth having both delegation() and
delegation_ref()

It looks like we could have delegation return a ref, and then callers .clone()
if they need to?

Since we're backporting this it's probably better to add delegation_ref() for
now to minimize the diff, but something to keep in mind for the future.

Reply:
Totally agree but didnt want to break the sdk api